### PR TITLE
Add findBoneByName() to Skeleton

### PR DESCRIPTION
**Pull request by takahirox**
This PR adds an useful method `findBoneByName()` to `Skeleton`.

At least `PropertyBinding` uses and `GLTFExporter` would use #12975

Do you folks think `SkinnedMesh` also should have that method?

### CODE REVIEW

**Comment by mrdoob** [comment]
> Do you folks think `SkinnedMesh` also should have that method?

That I don't know, but that function is indeed much more useful in `Skeleton` than in `PropertyBinding`.

**Review by donmccurdy — APPROVED** [review]
Nice, looks good! Putting it on SkinnedMesh would be OK too... can't decide which I prefer, so either way.

**Comment by takahirox** [comment]
Then, let only `Skeleton` have the method for now.
Let's consider `SkinnedMesh` when necessary.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Actually, lets rename this to `getBoneByName`. To align with `Object3D`'s `getObjectByName`.

**Comment by haroldiedema** [comment]
> Do you folks think SkinnedMesh also should have that method?

This would be great for "merging" animation files in a model file.

Practical use case:

My `.dae` files can't have multiple animations inside them, so I have one model file and a bunch of `.dae` files that are animations.

I currently hack around this by iterating recursively through all the bones in the animation file and re-assign the references to the model file, so the `AnimationMixer` knows how to play the animation on the model.

```javascript
let model = this.loader.load('my-model.dae'),
    anim1 = this.loader.load('my-anim1.dae');

this._injectAnimation("walking", anim1, model);

// Some time later...
this.animator.play('walking'); // will perform animation from "my-anim1.dae" on "my-model.dae".
```

snippet from my code that makes "the magic happen":
```javascript
    /**
     * Adds animation sources to the main model.
     *
     * @private
     * @param {String}    name
     * @param {Object3D}  object
     * @param {Animation} animation
     */
    _injectAnimation (name, object, animation)
    {
        animation.tracks.forEach((track) => {
            let chunks = track.name.split('.'),
                uuid   = chunks[0],
                part   = chunks[1];

            let target_uuid = this._findUUIDByName(this._model, this._findNameByUUID(object, uuid));
            if (typeof target_uuid === 'undefined') {
                throw new Error('Unable to retarget animation track "' + track.name + '". There is no object with UUID "' + uuid + '" in the model.');
            }

            track.name = target_uuid + '.' + part;
        });

        let action  = this._mixer.clipAction(animation);
        action.name = name;

        return action;
    }

    /**
     * Finds a name from the given object by the given UUID.
     * Used by _injectAnimation() to retarget animation tracks to bones of the model.
     *
     * @private
     * @param   {Object3D} object
     * @param   {String}   UUID
     * @returns {String}
     */
    _findNameByUUID (object, UUID)
    {
        let found;
        object.traverse((obj) => {
            if (found) return;
            if (obj.uuid === UUID) found = obj.name;
        });
        return found;
    }

    /**
     * Finds a UUID from the given object by the given name.
     * Used by _injectAnimation() to retarget animation tracks to bones of the model.
     *
     * @param   {Object3D} object
     * @param   {String}   name
     * @returns {*}
     * @private
     */
    _findUUIDByName (object, name)
    {
        let found;
        object.traverse((obj) => {
            if (found) return;
            if (obj.name === name) {
                found = obj.uuid;
            }
        });
        return found;
    }
```

File: src/objects/Skeleton.js [hunk 1]
```diff
@@ -152,6 +152,24 @@ Object.assign( Skeleton.prototype, {
...
+
+			if ( bone.name === name ) {
+
+				return bone;
+
+			}
+
+		}
+
+		return undefined;
```
Comment (mrdoob) [inline src/objects/Skeleton.js]: I think this is is what a function returns by default.

**Comment by takahirox** [comment]
Updated.
I renamed and also let it return `undefined` if the target bone isn't found, to apply with `Object3D.getObjectByName`'s behavior.

**Comment by mrdoob** [comment]
> also let it return `undefined` if the target bone isn't found, to apply with `Object3D.getObjectByName`'s behavior.

I see!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**